lib/qb_integration/services/line.rb

```diff
@@ -130,8 +130,11 @@ def build_from_adjustments(account = nil)
           adjustment[:name] = adjustment["name"]
           adjustment[:sku] = sku
 
+          found_item = item_service.find_or_create_by_sku(adjustment, account)
+          raise RecordNotFound.new("Adjustment Item #{adjustment[:name]} with SKU #{adjustment[:sku]} not found in QuickBooks") unless found_item
```

Comment:
This looks good! I'm guessing there are cases where the find or create doesn't return an item? My only suggestion might be to have this error in the item service instead but it might be better here because you can say it was an adjustment specifically

Reply:
Hmm that's a really good point. I'll look to see where else we call this. Wondering if there is any place where it's ok for the method to return nil (AKA where we wouldn't want to raise an error)

Reply:
Ahh, ok. This is probably safer then and its usually the default behavior anyways. Naming convention-wise, only methods that end with a bang `!` are supposed to throw errors when used

Reply:
Wow - I learned something new today. Didn't know that!

Reply:
Yeah, technically, they're supposed to mean that you're mutating something in the state but when you use it in rails, bangs throw an error if they fail a validation while the non-bang version just doesn't save